hack/lib/common-var-init.sh

```diff
@@ -215,7 +215,24 @@ REUSE_CERTS=${REUSE_CERTS:-false}
 #
 # Avoid to create loop error at https://coredns.io/plugins/loop/#troubleshooting
 # when create coredns pod
-RESOLV_CONF=${RESOLV_CONF:-"/run/systemd/resolve/resolv.conf"}
+RESOLV_CONF=${RESOLV_CONF:-}
+if [ -z ${RESOLV_CONF} ] && [ -z ${DISABLE_NETWORK_SERVICE_SUPPORT} ]
+then
+  if [ -f "/run/systemd/resolve/resolv.conf" ]; then
+    RESOLV_CONF=${RESOLV_CONF:-"/run/systemd/resolve/resolv.conf"}
+  else
+    if [ -z ${UPSTREAM_DNS_IP:-} ]
+    then
+      echo "please specify the upstream DNS ip address in env var UPSTREAM_DNS_IP" >&2
+      exit 1
+    fi
+    RESOLV_CONF=$(mktemp -p /tmp)
+    cat << EOF > ${RESOLV_CONF}
+nameserver ${UPSTREAM_DNS_IP}
```

Comment:
Try to run arktos-up, got message "please specify the upstream DNS ip address in env var UPSTREAM_DNS_IP". There need to be a default or suggested setting. This is not user friendly.

Reply:
on AWS Ubuntu 16.04, can change nameserver from 127.0.0.53 to 8.8.8.8 in /etc/resolv.conf then create new temporary file /tmp/resolv.conf when the file /run/systemd/resolve/resolv.conf is not available.
```
sed -e "s/127.0.0.53/8.8.8.8/g" /etc/resolv.conf > /tmp/resolve.conf
```

Then codes can be changed to the following:

```
RESOLV_CONF=${RESOLV_CONF:-}
if [ -z ${RESOLV_CONF} ] && [ -z ${DISABLE_NETWORK_SERVICE_SUPPORT} ]
then
  if [ -f "/run/systemd/resolve/resolv.conf" ]; then
    RESOLV_CONF=${RESOLV_CONF:-"/run/systemd/resolve/resolv.conf"}
  else
     sed -e "s/127.0.0.53/8.8.8.8/g" /etc/resolv.conf > ./tmp/resolve.conf
     RESOLV_CONF=${RESOLV_CONF:-"/tmp/resolv.conf"}
  fi
```